src/package-resolver.js

```diff
@@ -23,11 +24,12 @@ export type ResolverOptions = {|
 |};
 
 export default class PackageResolver {
-  constructor(config: Config, lockfile: Lockfile) {
+  constructor(config: Config, lockfile: Lockfile, resolutions: ?Resolutions) {
     this.patternsByPackage = map();
     this.fetchingPatterns = map();
     this.fetchingQueue = new BlockingQueue('resolver fetching');
     this.patterns = map();
+    this.resolutions = resolutions || new Resolutions(config);
```

Comment:
Note that they have different semantics. The default parameter will only work if the argument is omitted. If the argument is falsy, it will be null in the end:

```javascript
const fn = (arg = 42) => arg;
fn(null); // null

const fn = arg => arg || 42;
fn(null); // 42
```

It depends on what you are looking for.